src/foam/dao/ResetDAODecorator.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'ResetDAODecorator',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: `DAO decorator that resets the given dao and clears it's cache after put.`,
+  
+  requires: [
+    'foam.dao.AbstractDAO',
+    'foam.dao.TTLSelectCachingDAO'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'daoKey'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'put_',
+      code: function(x, obj) {
+        return this.delegate.put_(x, obj).then(() => {
```

Comment:
@kgrgreer can we use a framed listener somehow here so multiple puts reset once - thinking of the wizard - if one were to use this there.

Reply:
I don't think that would be helpful because we don't fill the cache until you perform the same query anyway, so multiple purges before a select() wouldn't be harmless, and second we would typically do the select() right after the put() anyway, so if we had introduced a delay we would have gotten stale data.